homeassistant/components/fronius/sensor.py

```diff
@@ -736,5 +750,13 @@ def __init__(
         """Set up an individual Fronius storage sensor."""
         self._name_extension = f"Storage {solar_net_id}"
         super().__init__(coordinator, key, solar_net_id)
-
-        self._attr_unique_id = f'{self._device_data()["serial"]["value"]}-{key}'
+        storage_data = self._device_data()
+
+        self._attr_unique_id = f'{storage_data["serial"]["value"]}-{key}'
+        self._attr_device_info = DeviceInfo(
```

Comment:
No I have no real-world data for batteries since I don't have one. 
On discord someone offered to send me responses of his system when his battery is installed about end of November.